src/Common/AsyncTaskExecutor.h

```diff
@@ -124,48 +123,6 @@ class AsyncTaskExecutor
     std::unique_ptr<AsyncTask> task;
 };
 
-/// Simple implementation for fiber local variable.
-template <typename T>
-struct FiberLocal
```

Comment:
This implementation from previous fix is not good for 2 reasons:
1) It works only when fibers are executed in one thread. But sometimes it can happen that fiber was started in one thread, suspended and resumed in another thread (for example during query cancellation).
2) It doesn't clean instances created for fibers when fiber is destroyed (I just didn't think about it, my fault).